WordPress/Classes/ViewRelated/Views/List/NotificationsList/AvatarsView.swift

```diff
@@ -0,0 +1,144 @@
+        case triple(URL, URL, URL)
+
+        var diameter: CGFloat {
+            switch self {
+            case .single:
+                return 40
+            case .double:
+                return 32
+            case .triple:
+                return 28
+            }
+        }
+
+        var leadingOffset: CGFloat {
+            switch self {
+            case .single:
+                return 0
+            case .double:
+                return 5
+            case .triple:
+                return Length.Padding.split/2
+            }
+        }
+    }
+
+    private let style: Style
+
+    init(style: Style) {
+        self.style = style
+    }
+
+    var body: some View {
+        switch style {
+        case let .single(primaryURL):
+            avatar(url: primaryURL)
+        case let .double(primaryURL, secondaryURL):
+            doubleAvatarView(
+                primaryURL: primaryURL,
+                secondaryURL: secondaryURL
+            )
+        case let .triple(primaryURL, secondaryURL, tertiaryURL):
+            tripleAvatarView(
+                primaryURL: primaryURL,
+                secondaryURL: secondaryURL,
+                tertiaryURL: tertiaryURL
+            )
+        }
+    }
+
+    private func avatar(url: URL) -> some View {
+        AsyncImage(url: url) { image in
+            image.resizable()
+        } placeholder: {
+            Color.DS.Background.secondary
+        }
+        .frame(width: style.diameter, height: style.diameter)
+        .clipShape(Circle())
+    }
+
+    private func doubleAvatarView(primaryURL: URL, secondaryURL: URL) -> some View {
+        ZStack {
+            HStack {
+                avatar(url: secondaryURL)
+                Spacer().frame(width: Constants.doubleAvatarHorizontalOffset)
+            }
+            HStack {
+                Spacer().frame(width: Constants.doubleAvatarHorizontalOffset)
+                avatar(url: primaryURL)
+                    .avatarBorderOverlay()
+            }
+        }
+        .frame(height: style.diameter)
+    }
+
+    private func tripleAvatarView(
+        primaryURL: URL,
+        secondaryURL: URL,
+        tertiaryURL: URL
+    ) -> some View {
+        ZStack {
+            HStack {
+                avatar(url: tertiaryURL)
+                Spacer().frame(width: Length.Padding.medium)
+            }
+            VStack {
+                avatar(url: secondaryURL)
+                    .avatarBorderOverlay()
+                Spacer().frame(height: Length.Padding.large)
+            }
+            HStack {
+                Spacer().frame(width: Length.Padding.medium)
+                avatar(url: primaryURL)
+                    .avatarBorderOverlay()
+            }
+        }
+        .frame(height: Constants.tripleAvatarViewHeight)
+    }
+}
+
+private extension View {
+    func avatarBorderOverlay() -> some View {
+        self.overlay(
+            Circle()
+                .stroke(Color.DS.Background.primary, lineWidth: 1)
```

Comment:
[Suggestion] I think this should be customizable so it exactly matches the background color. For example, if the background is pure black, the stroke should be pure black as well.

Reply:
That's a great point!

Reply:
Resolved in #https://github.com/wordpress-mobile/WordPress-iOS/pull/22564/commits/1f05244688957cc49b6756c34b3a78da0c07ef91